src/l2_tracker/events.h

```diff
@@ -16,7 +15,15 @@ using namespace std::literals;
 
 namespace eth::event {
 
-struct L2StateChange {};
+struct L2StateChange {
+    uint64_t l2_height = 0;
```

Comment:
Thoughts on having a domain-separation tag like we do in BLS schemes and other chains do differentiate the hashes per network type?

Reply:
Can you elaborate on what this would look like?

Reply:
For example the BLS tag is `H("<command>" || chain_id || contract_address)`

It's probably just enough to do 

```
H(l2_height || network_chain_id || tx_label || tx_blob)`

# With tx_label being "ServiceNodeRemovalRequest", "ServiceNodeRemoval" or "NewServiceNode".
```

It's not too annoying to smuggle in the network chain ID into the tx_extra here as there's only one place to construct these L2 state changes.

Reply:
I'd added the chainId into the L2StateChange base class now, and added it to serialization (commit: `Add chainId to events; check chainId on startup`, 2482b347b1c9b770cd31a957843e16aaf60ffa2f).

The separate hashing isn't necessary: the tx hash already applies over the serialized tx data, which (now) contains (serialized) chain id, and there's a unique type value in the serialized transaction for each event type, so that a tx hash already depends on each of l2_height, chain_id, event type, and other event parameters.